### PR TITLE
Use project.json for .csproj package references via kpm list

### PR DESCRIPTION
**Pull request by bricelam**
This gets rid of our silly `<PackageReference>` elements and makes `project.json` the source of truth for both csproj and kproj-based solutions. It also enables us to use some of the newer contracts like `System.IO.FileSystem` from the csproj-based solution.

Unfortunately, there is a bug in .NET Framework 4.6 that prevents the tests from running in the csproj-based solution. We probably want to wait until the fix is released publicly before merging.

### CODE REVIEW

File: tools/EntityFramework.targets [hunk 1]
```diff
@@ -2,83 +2,49 @@
...
-        Directory.CreateDirectory(Path.GetDirectoryName(TargetsFile));
         using (var writer = new StreamWriter(TargetsFile, append: false))
         {
             writer.WriteLine("<Project xmlns='http://schemas.microsoft.com/developer/msbuild/2003'>");
             writer.WriteLine("  <ItemGroup>");
 
-            var installedPackages = Directory.EnumerateDirectories(packagesFolder).Select(d => Path.GetFileName(d))
-                .ToArray();
-            foreach (var package in PackageReferences)
+            foreach (var assemblyPath in File.ReadLines(PackageReferences).Where(File.Exists))
```
Comment (bricelam) [inline tools/EntityFramework.targets]: @troydai and I decided that using `File.Exists()` is a good contract not to break for the output of `kpm list --assemblies`.

Comment (troydai) [inline tools/EntityFramework.targets]: :+1:

**Comment by bricelam** [comment]
This depends on aspnet/Universe#167.

File: src/EntityFramework.Commands/EntityFramework.Commands.csproj [hunk 2]
```diff
@@ -10,15 +11,18 @@
     <AppDesignerFolder>Properties</AppDesignerFolder>
     <RootNamespace>Microsoft.Data.Entity.Commands</RootNamespace>
     <AssemblyName>EntityFramework.Commands</AssemblyName>
-    <TargetFrameworkVersion>v4.5.1</TargetFrameworkVersion>
+    <DefaultLanguage>en-US</DefaultLanguage>
     <FileAlignment>512</FileAlignment>
+    <ProjectTypeGuids>{786C830F-07A1-408B-BD7F-6EE04809D6DB};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
+    <TargetFrameworkProfile>Profile7</TargetFrameworkProfile>
+    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
```
Comment (bricelam) [inline src/EntityFramework.Commands/EntityFramework.Commands.csproj]: Note to self, we should leave this as net451 so we get test coverage on that framework.

**Comment by MrTomWhite** [comment]
:+1: This is an awesome fix.

**Comment by bricelam** [comment]
Closing for now. Will re-open when unblocked

**Comment by bricelam** [comment]
New approach: Same thing, but net45 projects instead of .NET Core 5

**Pull request closed without merging**